packages/lodestar-validator/src/api/impl/rest/validator/validator.ts

```diff
@@ -37,25 +40,23 @@ export class RestValidatorApi implements IValidatorApi {
     return result;
   }
 
-  public async getAttesterDuties(epoch: number, validatorPubKeys: Buffer[]): Promise<ValidatorDuty[]> {
+  public async getAttesterDuties(epoch: Epoch, validatorPubKeys: BLSPubkey[]): Promise<AttesterDuty[]> {
     const hexPubKeys = validatorPubKeys.map(toHexString);
     const url = `/duties/${epoch.toString()}/attester?validator_pubkeys=${JSON.stringify(hexPubKeys)}`;
     const responseData = await this.client.get<Json[]>(url);
-    return responseData.map(value => this.config.types.ValidatorDuty.fromJson(value));
+    return responseData.map(value => this.config.types.AttesterDuty.fromJson(value));
   }
 
   public async publishAggregatedAttestation(
-    aggregatedAttestation: Attestation,
-    validatorPubkey: BLSPubkey,
-    slotSignature: BLSSignature
+    aggregate: AggregateAndProof,
   ): Promise<void> {
     return this.client.post(
-      `/aggregate?validator_pubkey=${toHexString(validatorPubkey)}&slot_signature=${toHexString(slotSignature)}`,
-      this.config.types.Attestation.toJson(aggregatedAttestation)
+      "/aggregate_and_proof",
+      [this.config.types.AggregateAndProof.toJson(aggregate)]
```

Comment:
I'm curious about the reason to have an array here instead of a single AggregateAndProof

Reply:
probably because one may want to run multiple validators in a single client?

Reply:
yeah, afaik lighthouse is relying heavily on batch requests